pkg/api/customization/cluster/actions_cis.go

```diff
@@ -25,9 +23,22 @@ const (
 )
 
 func (a ActionHandler) runCisScan(actionName string, action *types.Action, apiContext *types.APIContext) error {
-	var clusterForAccessCheck mgmtclient.Cluster
 	var err error
 
+	canCreateClusterScanObject := func() bool {
+		return apiContext.AccessControl.CanDo(
+			v3.ClusterScanGroupVersionKind.Group,
+			v3.ClusterScanResource.Name,
+			"create",
+			apiContext,
+			nil,
+			apiContext.Schema,
+		) == nil
+	}
+	if !canCreateClusterScanObject() {
+		return httperror.NewAPIError(httperror.PermissionDenied, "can not run security scan")
+	}
+
```

Comment:
We should also check for ability to update cluster since this action will also update the cluster object.
example: https://github.com/rancher/rancher/blob/393dadee4196a2ccbe31bde711df6a50f29a32ca/pkg/api/customization/cluster/actionhandler.go#L97

Reply:
Will do